examples/udp/transport.cpp

```diff
@@ -0,0 +1,180 @@
+#include "caf/ipv4_address.hpp"
+#include "caf/log/net.hpp"
+
+#include <csignal>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+using namespace caf;
+using namespace caf::net;
+
+using namespace std::literals;
+
+namespace {
+
+std::atomic<bool> shutdown_flag;
+
+// Implementation of logger interface for our custom component "app".
+namespace app {
+
+constexpr std::string_view component = "app";
+
+template <class... Ts>
+void debug(caf::format_string_with_location fmt_str, Ts&&... args) {
+  caf::logger::log(caf::log::level::debug, component, fmt_str,
+                   std::forward<Ts>(args)...);
+}
+
+template <class... Ts>
+void info(caf::format_string_with_location fmt_str, Ts&&... args) {
+  caf::logger::log(caf::log::level::info, component, fmt_str,
+                   std::forward<Ts>(args)...);
+}
+
+template <class... Ts>
+void error(caf::format_string_with_location fmt_str, Ts&&... args) {
+  caf::logger::log(caf::log::level::error, component, fmt_str,
+                   std::forward<Ts>(args)...);
+}
+
+} // namespace app
+
+} // namespace
+
+// -- configuration setup ------------------------------------------------------
+
+struct config : actor_system_config {
+  config() {
+    opt_group{custom_options_, "global"}
+      .add<std::string>("address,a", "Address to send payload to")
+      .add<std::string>("port,p", "Port to send payload to")
+      .add<uint16_t>("listen,l", "Port to listen for incoming data");
+  }
+};
+
+// -- utility functions --------------------------------------------------------
+
+// Read lines from standard input. Run as detached actor.
+void reader(event_based_actor* self, actor sink, ip_endpoint dest) {
+  std::string line;
+  while (std::getline(std::cin, line))
+    self->send(sink, dest, line);
+  // Stop the application when reading EOF.
+  shutdown_flag = true;
+}
+
+// Print lines to standard output.
+behavior printer(event_based_actor* self) {
+  return {[](ip_endpoint ep, std::string message) {
+            app::info("caf", "received {} bytes from {}:{}", message.size(),
+                      ep.address(), ep.port());
+            std::cout << message << std::endl;
+          },
+          [self](const message&) {
+            app::error("caf", "received unexpected message");
+            self->quit();
+          }};
+}
+
+// Send payload over udp.
+error send_payload(actor_system& sys, std::string addr, uint16_t port) {
+  auto tg = logger::trace("caf", "Trying to send data to {}:{}", addr, port);
+  auto print_actor = sys.spawn(printer);
+  // Simplified case, we use only the first resolved address.
+  auto addrs = ip::resolve(addr);
+  if (addrs.empty())
+    return make_error(sec::runtime_error, "No resolved ip address");
+  auto dest = ip_endpoint{addrs[0], port};
+  auto local = ip_endpoint{make_ipv4_address(127, 0, 0, 1), 0};
+  auto sock = make_udp_datagram_socket(local);
+  if (!sock) {
+    std::cerr << "*** failed to open udp with: " << std::endl;
```

Comment:
This is also inconsistent. Either we always use `stderr` or we use `app::error`. I'd be in favor of the latter.